app/docs/0.11.x/clustering.md

```diff
@@ -4,135 +4,54 @@ title: Clustering Reference
+registered in the datastore. Periodically all nodes will check for updates
+here and apply the changes.
 
-A Kong node can be in four different states:
+Kong clustering settings are specified in the configuration file at the
+following entries:
 
-* `active`: the node is active and part of the cluster.
-* `failed`: the node is not reachable by the cluster.
-* `leaving`: a node is in the process of leaving the cluster.
-* `left`: the node has gracefully left the cluster.
+* [db_update_frequency][db_update_frequency]
+* [db_update_propagation][db_update_propagation]
 
-When a node is `failed`, you need to manually remove it from the cluster.
-
-## 5. Removing a failed node
-
-Everytime a new Kong node is stopped, that node will try to gracefully remove itself from the cluster. When a node has been successfully removed from a cluster, its state transitions from `alive` to `left`.
-
-To gracefully stop and remove a node from the cluster just execute the `kong quit` or `kong stop` CLI commands.
-
-When a node is not reachable for whatever reason, its state transitions to `failed`. Kong will automatically try to re-join a failed node just in case it becomes available again. You can exclude a failed node from the cluster in two ways:
-
-* Using the [`kong cluster force-leave`][cli-cluster] CLI command
-* Using the [API][cluster-api-remove]. 
-
-Check the [Node Failures](#node-failures) paragraph for more info.
-
-## 6. Edge-case scenarios
-
-The implementation of the clustering feature of Kong is rather complex and may involve some edge case scenarios.
-
-#### Asynchronous join on concurrent node starts
-
-When multiple nodes are all being started simultaneously, a node may not be aware of the other nodes yet because the other nodes didn't have time to write their data to the datastore. To prevent this situation Kong implements by default a feature called "asynchronous auto-join".
-
-Asynchronous auto-join will check the datastore every 3 seconds for 60 seconds after a Kong node starts, and will join any node that may appear in those 60 seconds. This means that concurrent environments where multiple nodes are started simultaneously it could take up to 60 seconds for them to auto-join the cluster.
-
-#### Automatic cache purge on join
-
-Every time a new node joins the cluster, or a failed node re-joins the cluster, the in-memory cache for every node is purged and all the data is forced to be re-fetched from the datastore. This is to avoid inconsistencies between the data that has already been invalidated in the cluster, and the data stored on the node.
-
-This also means that after joining the cluster the new node's performance will be slower until the data has been re-cached into the local memory.
-
-#### Node failures
+<div class="alert alert-warning">
+  Please note that both Kong itself as well as the Cassandra datastore follow
+  an `eventual consistency` model. And hence changes need some time to be
+  effectuated across the cluster.
```

Comment:
Correct, it's not. However, the fact that Kong uses data stores that support both models should be noted. If anything, let's note that, because Postgres is not typically deployed in an eventually consistent model, the behaviors between Kong <-> psql might be different than one might see with Kong <-> c*